### DISCUSSION TITLE
Adaptive updates with Transfer-Encoding instead Http streamign

### DISCUSSION THREAD

**Discussion by tcikel** [rauc/rauc#1265] [Q&A]
Hi,

I would like to use ThingsBoard as server instead of Hawkbit.
I would like to write a client similar to rauc-hawkbit-updater. I understand that for adaptive updates the server needs http streaming. From my limited testing on ThingsBoard demo client I find out that it doesnt support http streaming. Although it does support Transfer-Encoding which is kinda similar, it also allows to download specific chunks of files. Could it be possible tu use RAUC adaptive updates also with Transfer-Encoding? Or is there any plan to implement it in future? 

Thank you.

**Comment by ejoerns** [comment]
@tcikel Having a client interacting with thingsboard would be interesting I guess.

Do they fully ignore/deny HTTP range requests or is it more like you hit some rate limiting when trying to download data?

I have no real idea of Transfer-Encoding, but for me, it does not look like you can request arbitrary data ranges from a given artifact on the server with that, or could you? This however would be required to make streaming work.

Just for the theoretical case that it would work, this would then require writing an alternative NBD server in RAUC that uses Transfer-Encoding instead of HTTP Range requests.

Note that rauc-hawkbit-updater has no idea about streaming. So you could still write a client interfacing with RAUC and ThingsBoard, but omit the streaming feature. But I assume this would not be sufficient for your use case, right?

**Reply by jluebbe** [reply]
If ThingsBoard can send a HTTP redirect, you could simply host the bundle on a server that supports range requests. The current streaming client already follows redirects.
